src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -1219,6 +1252,65 @@ internal async void OnTimer(object sender, System.Timers.ElapsedEventArgs e)
             }
         }
 
+        internal void OnInvocationTimer(object sender, System.Timers.ElapsedEventArgs e)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            try
+            {
+                CheckAndRecycleFaultyLanguageWorker();
+            }
+            catch
+            {
+                // Don't allow background execptions to escape
+                // E.g. when a rpc channel is shutting down we can process exceptions
+            }
+            try
+            {
+                _invocationTimer.Start();
+            }
+            catch (ObjectDisposedException)
+            {
+                // Specifically ignore this race - we're exiting and that's okay
+            }
+        }
+
+        internal void CheckAndRecycleFaultyLanguageWorker()
```

Comment:
Sorry If I am comprehending this incorrectly. :) 

I wonder this method belongs to this class? Recycling means killing the instance of this (GrpcWorkerChannel) class. Correct? If so, shouldn't we do this operation from another class which is still alive ? Perhaps some class which is monitoring the metric data for each worker and then initiate the recycle operation form there.

Reply:
Her intent was that this is a placeholder for the work we're ultimately planning. However, because we're only logging this first round and not recycling anything, I think we can simplify, remove all the bits about recycling, thresholds, etc. and just log. Rename this method to LogInvocationMetrics

Reply:
Renamed method to `IdentifyFaultyLanguageWorker()` if this looks okay.